src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -634,6 +634,41 @@ TransactionManager::get_extents_if_live(
   });
 }
 
+TransactionManager::pin_to_extent_by_type_ret
+TransactionManager::pin_to_extent_by_type(
```

Comment:
because pin_to_extent_by_type is called only when the extent is not in Cache, which means the corresponding pointer in the lba leaf node must be nullptr.

Reply:
Do you mean this code?
https://github.com/ceph/ceph/blob/f5c5009eaba4a83cc9a8b031636bfb9af00ad83d/src/crimson/os/seastore/transaction_manager.cc#L551-L554

That doesn't seem to me that "the extent is not in Cache", the extent can be in cache but has a different length.

Another reason is that `pin_to_extent_by_type()` doesn't need to be different from `pin_to_extent()` besides the additional `type`. And the following `cache->get_extent_by_type()` can also use the `need_check_trans=false` version, right?

Reply:
> That doesn't seem to me that "the extent is not in Cache", the extent can be in cache but has a different length.

This "extent->get_length() == len" condition is always true and not necessary, I've just submitted a PR #50704 to remove it, please take a look.



> Another reason is that `pin_to_extent_by_type()` doesn't need to be different from `pin_to_extent()` besides the additional `type`. And the following `cache->get_extent_by_type()` can also use the `need_check_trans=false` version, right?
> 
> If that's true, probably better to move it back to .h after `pin_to_extent()`?

I'll try this.